polyfactory/utils/predicates.py

```diff
@@ -131,6 +139,16 @@ def is_any_annotated(annotation: Any) -> bool:
     )
 
 
+def is_type_alias(annotation: Any) -> TypeGuard[TypeAliasType]:
+    """Determine if the given type annotation is a PEP 695 type alias.
+
+    :param annotation: A type annotation.
+
+    :returns: A boolean
+    """
+    return isinstance(annotation, TypeAliasType)
```

Comment:
This would break the other use case. Even if `TypeAliasType` is available in `typing`, users might still import it from `typing_extensions`. So this code should always check both if both are available